src/ansible_navigator/configuration_subsystem/utils.py

```diff
@@ -0,0 +1,174 @@
+"""Utilities related to the configuration subsystem."""
+
+import sys
+
+from types import SimpleNamespace
+from typing import Dict
+from typing import List
+from typing import NamedTuple
+from typing import Union
+
+from .definitions import ApplicationConfiguration
+from .definitions import CliParameters
+from .definitions import Constants as C
+from .definitions import SettingsEntry
+
+
+class HumanReadableEntry(NamedTuple):
+    """Data structure for a setting entry."""
+
+    #: The possible values
+    choices: List
+    #: The CLI parameters, short and long
+    cli_parameters: Dict[str, str]
+    #: The path to the current settings file
+    current_settings_file: str
+    #: The current, effective value
+    current_value: Union[bool, Dict, str, List]
```

Comment:
We talked about that, but really the type for "current_value" should be part of the configuration subsystem defs and shouldn't need to be defined here....  

I'd rather not see it added outside this file in this PR, think it should be here for now? Or based on our Generic polymorphic type conversation should it wait?

Reply:
I think ultimately we'll want this class (especially if it's a dataclass as discussed a few lines up) to be polymorphic over the type of `current_value`. So instead of `HumanReadableEntry`, it'll be `HumanReadableEntry[T]` where `T` is the type of the value of the setting.

But I'd probably want to do this after the other changes we discussed. For now just doing a type alias seems like a nice way to remove some duplication until that other work lands.